firestore.rules

```diff
@@ -1965,6 +1966,60 @@ service cloud.firestore {
       allow delete: if checkDeleteQueueEntity(request.auth);
     }
 
+    // Roles implementation
+    // -----------------------------
+
+    function getRoleAfter(id, index) {
+      return getAfter(/$(databasePath())/UserRoles/$(id)/Roles/$(index))
+    }
+
+    function getRoleBefore(id, index) {
+      return get(/$(databasePath())/UserRoles/$(id)/Roles/$(index))
+    }
+
+    function roleExistsAfter(id, index) {
+      return existsAfter(/$(databasePath())/UserRoles/$(id)/Roles/$(index))
+    }
+
+    function roleExistsBefore(id, index) {
+      return exists(/$(databasePath())/UserRoles/$(id)/Roles/$(index))
+    }
+
+    function validateRole(resource) {
+      return isString(resource.data.roleName) &&
+      resource.id == resource.data.roleName &&
+      userExistsBefore(resource.data.userId) &&
+      (resource.data.roleName == 'admin' || resource.data.roleName == 'hil')
+    }
+
+    function checkReadRole(auth, resource) {
+      return currentUserCanReadPublicEntity(auth, resource, false)
```

Comment:
These entities should probably be private read. Don't want a user to be able to scan for admin account ids.